api/envoy/extensions/access_loggers/fluentd/v3/fluentd.proto

```diff
@@ -85,4 +86,9 @@ message FluentdAccessLogConfig {
   // as specified in the :ref:`RetryOptions <envoy_v3_api_msg_extensions.access_loggers.fluentd.v3.FluentdAccessLogConfig.RetryOptions>`
   // configuration.
   RetryOptions retry_options = 7;
+
+  // Specifies a collection of Formatter plugins that can be called from the access log configuration.
+  // See the formatters extensions documentation for details.
+  // [#extension-category: envoy.formatter]
+  repeated config.core.v3.TypedExtensionConfig formatters = 8;
```

Comment:
I re-checked the implementation, I think maybe we can reuse the `SubstitutionFormatString` here directly? WDYT?

Reply:
Not sure how? the ``SubstitutionFormatString`` API has options to configure text/json/from source which isn't related to msgpack. The msgpack itself needs to be defined with a struct. Can you elaborate what you had in mind?

Reply:
Because your current `record` + `formatters` basically equals to `json_format` + `formatters` in `SubstitutionFormatString`.

IMO, although `SubstitutionFormatString` contains some API that you needn't, but it should still be OK to reuse it directly. And you can make additional restriction or validation in your fluend implementation (like only allow `json_format` be used.)

Reply:
okay